schema/types/acname.go

```diff
@@ -72,3 +74,22 @@ func (n *ACName) UnmarshalJSON(data []byte) error {
 func (n *ACName) MarshalJSON() ([]byte, error) {
 	return json.Marshal(n.String())
 }
+
+// SanitizeACName replaces every invalid ACName character in s with a dash
+// making it a legal ACName string. If the character is an upper case letter it
+// replaces it with its lower case. It also removes illegal edge characters
+// (hyphens, periods and slashes).
+func SanitizeACName(s string) (string, error) {
+	s = strings.ToLower(s)
+	invalidChars := regexp.MustCompile("[^a-z0-9./-]")
```

Comment:
Please initialise these globally vars so that MustCompile would capture them at runtime initialisation